toqito/channel_props/tests/test_is_extremal.py

```diff
@@ -0,0 +1,132 @@
+"""Tests for the `is_extremal` function in the Toqito library."""
+
+import numpy as np
+import pytest
+
+from toqito.channel_ops.kraus_to_choi import kraus_to_choi
+from toqito.channel_props.is_extremal import is_extremal
+
+
+# Test cases for unitary channels (which are extremal)
+@pytest.mark.parametrize(
+    "phi, expected",
+    [
+        # Direct flat list of Kraus operators
+        ([np.array([[0, 1], [1, 0]])], True),
+        # Nested list of Kraus operators
+        ([[np.array([[0, 1], [1, 0]])]], True),
+    ],
+)
+def test_extremal_unitary_channel(phi, expected):
+    """Verify that unitary channels are correctly identified as extremal."""
+    assert is_extremal(phi) == expected
+
+
+# Test cases for non-extremal channels
+@pytest.mark.parametrize(
+    "phi, expected",
+    [
+        (
+            # Flat list: two identical Kraus operators.
+            [
+                np.sqrt(0.5) * np.array([[1, 0], [0, 1]]),
+                np.sqrt(0.5) * np.array([[1, 0], [0, 1]]),
+            ],
+            False,
+        ),
+        (
+            # Nested list version of the above.
+            [
+                [np.sqrt(0.5) * np.array([[1, 0], [0, 1]])],
+                [np.sqrt(0.5) * np.array([[1, 0], [0, 1]])],
+            ],
+            False,
+        ),
+    ],
+)
+def test_non_extremal_channel(phi, expected):
+    """Verify that non-extremal channels are correctly identified."""
+    assert is_extremal(phi) == expected
+
+
+# Test cases where the channel is provided as a Choi matrix
+@pytest.mark.parametrize(
+    "phi, expected",
+    [
+        (kraus_to_choi([np.array([[0, 1], [1, 0]])]), True),
+    ],
+)
+def test_choi_input(phi, expected):
+    """Verify that a channel provided as a Choi matrix is correctly processed."""
+    assert is_extremal(phi) == expected
+
+
+# Test example from Watrous's book (example 2.33)
+@pytest.mark.parametrize(
+    "phi, expected",
+    [
+        (
+            # Flat list representation.
+            [
+                (1 / np.sqrt(6)) * np.array([[2, 0], [0, 1], [0, 1], [0, 0]]),
+                (1 / np.sqrt(6)) * np.array([[0, 0], [1, 0], [1, 0], [0, 2]]),
+            ],
+            True,
+        ),
+        (
+            # Nested list representation.
+            [
+                [(1 / np.sqrt(6)) * np.array([[2, 0], [0, 1], [0, 1], [0, 0]])],
+                [(1 / np.sqrt(6)) * np.array([[0, 0], [1, 0], [1, 0], [0, 2]])],
+            ],
+            True,
+        ),
+    ],
+)
+def test_example_from_watrous(phi, expected):
+    """Test the example 2.33 from Watrous's *Theory of Quantum Information*."""
+    assert is_extremal(phi) == expected
+
+
+# Test depolarizing channel, which is non-extremal for d > 2 (here d=2)
+@pytest.mark.parametrize(
+    "phi, expected",
+    [
+        (
```

Comment:
For these two tests (for instance) they can be consolidated into one test that parameterizes over both of the examples. So, the function could be called `test_is_extremal` (for instance), and then you would have test cases in the parameterized section that would pertain to the depolarizing channel test case, the Watrous test case, etc. 

The same would apply to the test cases above this as well, they can likely be parameterized in a similar manner. 

Does that make sense?